internal/handler/impl/utils.go

```diff
@@ -20,3 +24,27 @@ func getXRHID(ctx echo.Context) (*identity.XRHID, error) {
 		return xrhid, nil
 	}
 }
+
+func sendPendoTrackEvent(ctx echo.Context, client pendo.Pendo, event string) {
+	c := ctx.Request().Context()
+	logger := app_context.LogFromCtx(c)
+
+	xrhid, err := getXRHID(ctx)
+	if err != nil {
+		logger.Error(err.Error())
+		return
+	}
+
+	track := pendo.TrackRequest{
+		AccountID: xrhid.Identity.OrgID,
+		Type:      "track",
+		Event:     event,
+		VisitorID: header.GetPrincipal(xrhid),
+		Timestamp: time.Now().UTC().Unix(),
+	}
```

Comment:
```suggestion
	track := pendo.TrackRequest{
		AccountID: xrhid.Identity.OrgID,
		Type:      pendoTypeTrack,
		Event:     event, // Core event column in Pendo?
		VisitorID: header.GetPrincipal(xrhid),
		Timestamp: time.Now().UTC().Unix(),
		URL:       ctx.Request().URL.String(), // not 100% sure if this would set the public or the in-cluster URL.
		UserAgent: ctx.Request().UserAgent(),  // not 100% sure, because the several reverse proxies
		IP:        GetRealIP(ctx), // Read real IP from headers
		// What additional properties to add if any?
		Properties: map[string]any{
			"my-property-1": true,
			"my-property-2": "some-value",
		},
		// What additional context to add if any?
		Context: map[string]any{
			"my-context-1": 5,
			"my-context-2": "another-value",
		},
		Title: "Insights IDMSVC Track Event",
	}
```

Reply:
I committed the suggestion by mistake; I have rolled back the branch to the state before the commit, and re-opened this suggestion; sorry for the inconveniences.